### PR TITLE
feat: Enable Zitadel Integration for API Authentication

### PR DESCRIPTION
**Pull request by wmoussa-gc**
# Summary | Résumé

Most of the work was ported from Bryan's original PR #3908. The only change made was in the management of Zitadel's admin key.

## Flow

https://github.com/user-attachments/assets/a354eee0-af30-4dba-a036-fa80aea0af90

## Machine Service Accounts

1 Form = 1 Account
image

# Test instructions | Instructions pour tester la modification

1. Switch on the feature flag for Zitadel
2. Go to Settings > API Access 
3. Click Generate Key

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

TBD

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [ ] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: lib/integration/zitadelConnector.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  createServiceAccountInterceptor,
+  ManagementServiceClient,
+} from "@zitadel/node/api";
+import { checkOne } from "@lib/cache/flags";
+import { ServiceAccount } from "@zitadel/node/credentials";
+
+let zitadelClient: ManagementServiceClient | null = null;
+
+const getZitadelSettings = async () => {
+  if (!process.env.ZITADEL_PROVIDER) throw new Error("No value set for Zitadel Provider");
```
Comment (wmoussa-gc) [inline lib/integration/zitadelConnector.ts:12]: you can find this value and the one below in 1Password

**Review by craigzour — COMMENTED** [review]
Looks good to me! Just want to flag that we still need to handle errors across the whole feature. We could create a ticket for it unless you want to tackle it now. What do you think?

**Comment by wmoussa-gc** [comment]
> We could create a ticket for it unless you want to tackle it now

I'd like to see a happy path working in staging, and then we can focus on stabilizing the code. What do you think?

**Comment by craigzour** [comment]
> > We could create a ticket for it unless you want to tackle it now
> 
> I'd like to see a happy path working in staging, and then we can focus on stabilizing the code. What do you think?

Sounds good to me! As long as we do not forget about it :)

**Review by craigzour — APPROVED** [review]
LGTM!

**Pull request merged**